distributed/client.py

```diff
@@ -2595,7 +2593,7 @@ def _graph_to_futures(
             self._send_to_scheduler(
                 {
                     "op": "update-graph-hlg",
-                    "hlg": dsk,
+                    "hlg": highlevelgraph_pack(dsk, self, keyset),
```

Comment:
Since `highlevelgraph_pack` is where we check that `dsk` doesn't contain `Future`s from other clients, we'll want to keep `highlevelgraph_pack(dsk, self, keyset)` above 

```python
futures = {key: Future(key, self, inform=False) for key in keyset}
```

where we create new `Future` objects. Otherwise, we'll end up creating `Future`s attached to a client that contain references to `Future`s on other clients, which isn't supported. See  `distributed/tests/test_client.py::test_mixing_clients` for a concrete example.

Reply:
You are absolutely right :)
I have moved it below `retries = {k: retries for k in dsk}` since  `dsk` are bytes after the `highlevelgraph_pack()` call.